### DISCUSSION TITLE
In memory dataset missing target label tensor.

### DISCUSSION THREAD

**Discussion by codyseally** [pyg-team/pytorch_geometric#3081] [Q&A] [answered]
I am first creating a pytorch data object (a general graph) like so: 

```
x = torch.tensor(node_features.to_numpy()[1:], dtype=torch.float)
edge_index_small = torch.tensor(alpha_edges, dtype=torch.long)
edge_index_clean, _, _ = remove_isolated_nodes(edge_index_small.t().contiguous())
y_val = y_label['label'].to_numpy()
y_tens = torch.tensor(y_val, dtype=torch.long)

data_small = Data(x=x, edge_index=edge_index_clean, y=y_tens)
torch.save(data_small, 'some_path')
```
where `y_label` is a pandas dataframe.

I then load this data object into an in-memory dataset, in another script. Part of the code below:

```
class MyGraph(InMemoryDataset):
    def __init__(self, root, transform=None, pre_transform=None):
        super(MetaGraph, self).__init__(root, transform, pre_transform)
        self.data, self.slices = torch.load(self.processed_paths[0])
        
    @property
    def raw_file_names(self):
        return ["mygraph"]

    @property
    def processed_file_names(self):
        return ['mygraph1.pt']

    def process(self):
        
        # read entries
        dataset_raw = torch.load(self.raw_paths[0])
        x = dataset_raw.x.float()
        edge_index = dataset_raw.edge_index.long()
        #edge_attr = dataset_raw.edge_attr.unsqueeze(0).t().float()
        print(dataset_raw.y)
        raise SystemExit
```

The `x` and `edge_index` attributes are fine, they are both torch tensors. However, when I print the labels (as in the above example), I get `None`.

```
dataset = MyGraph("path")
Processing...
None

An exception has occurred, use %tb to see the full traceback.

SystemExit
```
I have checked the labels before saving them into the data object, they are numeric, they seem to be fine, also the same length as `x`. I don't understand why it becomes `None` when I read the file. Please help.

**Comment by rusty1s** [comment, marked as answer]
I'm really not sure to be honest. The following works for me:
```python
import torch
from torch_geometric.data import Data

x = torch.randn(100, 128)
edge_index = torch.randint(0, 100, (2, 400), dtype=torch.long)
y = torch.randn(100)

data = Data(x=x, edge_index=edge_index, y=y)
torch.save(data, 'data.pt')

data = torch.load('data.pt')
print(data.x.shape)
print(data.edge_index.shape)
print(data.y.shape)
```

**Reply by codyseally** [reply]
Actually I will mark your reply as an answer, since I found my error in it. I replaced : 

`data = torch.load('data') `

with

`data = torch.load('data.pt')`

and the problem was solved. Seems you run into trouble if you don't specify the file extension.
